src/java.base/share/classes/jdk/internal/foreign/abi/AbstractLinker.java

```diff
@@ -134,6 +137,11 @@ public SystemLookup defaultLookup() {
         return SystemLookup.getInstance();
     }
 
+    @Override
+    public Map<String, MemoryLayout> canonicalLayouts() {
+        return CANONICAL_LAYOUTS_MAP;
```

Comment:
I'm not too sure on what this shared factories would look like. In the general case (e.g. fallback) it would have at least to specify the size of short/int/long/long long. Plus possibly, size_t. Basically everything besides float/double/char/bool. I'm not sure that would end up simpler than just having duplicated code with the layout creation?

Reply:
ok - I can try that